### PR TITLE
Update user guide contributing docs

### PR DESCRIPTION
**Pull request by lilyminium**
Related to #327

This PR removes some old section and adds some information on which files are Jupyter notebooks and which are not.

----
:books: Documentation preview :books:: https://mdanalysisuserguide--330.org.readthedocs.build/en/330/

### CODE REVIEW

File: doc/source/contributing_docs.rst [hunk 1]
```diff
@@ -34,6 +34,17 @@ will be rendered as:
         In [2]: x**3
         Out[2]: 8
 
+
+The ``nbsphinx`` extension lets you include Jupyter notebooks in the documentation.
+Everything in the `Examples <https://userguide.mdanalysis.org/stable/examples/README.html>`_
+or the `Analysis <https://userguide.mdanalysis.org/stable/examples/analysis/README.html>`_
+sections of the user guide is actually originally a Jupyter notebook.
```
Comment (IAlibay) [inline doc/source/contributing_docs.rst]:
```suggestion
sections of the user guide is written as a Jupyter notebook.
```

Comment (IAlibay) [reply doc/source/contributing_docs.rst]: Only a style thing, doesn't need accepting

File: doc/source/contributing_docs.rst [hunk 2]
```diff
@@ -169,52 +284,47 @@ environmental variable like this::
...
+=======================
 
+The below sections are optional, but may be helpful for more advanced users.
 
-Now you can build your pages with the ``make github`` macro in the ``UserGuide/doc/`` directory, which builds the files and copies them to the top level of your directory.
+Using pre-commit hooks
+""""""""""""""""""""""
 
-.. code-block:: bash
+Manually editing files can often lead to small inconsistencies: a whitespace here, a missing carriage return there. A tool called pre-commit can be used to automatically fix these problems, before a git commit is made. To enable the pre-commit hooks, run the following:
```
Comment (IAlibay) [inline doc/source/contributing_docs.rst]:
```suggestion
Manually editing files can often lead to small inconsistencies: a whitespace here, a missing carriage return there.
A tool called pre-commit can be used to automatically fix these problems, before a git commit is made.
To enable the pre-commit hooks, run the following:
```
Slight visual improvement on single long line.

File: doc/source/contributing_docs.rst [hunk 3]
```diff
@@ -231,23 +341,58 @@ Then, run the following command in the ``doc/`` directory:
...
+.. note::
 
-    .. code-block:: bash
+    This section documents how to render documentation on a fork without ReadTheDocs.
+    This is *generally unnecessary* and should only be done in cases where we
+    believe ReadTheDocs is not rendering our documentation properly.
+    For all other cases, please use the ReadTheDocs preview in pull requests.
 
-        pre-commit install
+It is often difficult to review Jupyter notebooks on GitHub, especially if you embed widgets and images.
```
Comment (IAlibay) [inline doc/source/contributing_docs.rst:351]: Should we be signposting to reviewnb here?

File: doc/source/contributing_docs.rst [hunk 4]
```diff
@@ -83,6 +75,133 @@ Using ``conda`` or similar (``miniconda``, ``mamba``, ``micromamba``), create a
...
+The user guide is largely composed of four different kinds of files:
+
+* :ref:`reStructuredText files <restructuredtext-files>` (``.rst``) which contain the text of the documentation
+* :ref:`Jupyter notebooks <jupyter-files>` (``.ipynb``) which contain code examples
+* Python scripts (``.py``) in the ``doc/source/scripts/`` directory which are used
+  for automatically generated documentation
+* Text files (``.txt``) which contain the automatically generated documentation.
+  These should not be edited directly as changes will not be included.
+  These are largely tables and lists of topology attributes for readers, writers,
+  and parsers
```
Comment (RMeli) [inline doc/source/contributing_docs.rst]:
```suggestion
  and parsers.
```

File: doc/source/contributing_docs.rst [hunk 5]
```diff
@@ -83,6 +75,133 @@ Using ``conda`` or similar (``miniconda``, ``mamba``, ``micromamba``), create a
...
+.. important::
+    
+    One of the neat things about ``nglview`` is the ability to interact with molecules via the viewer.
+    Many of our notebooks include cells that render MDAnalysis systems.
+    However, this creates very large files.
+    **We ask that you avoid saving the state of the viewer in the notebooks.**
+    We also ask in general that you leave NGLView cells commented out.
+
+Everything in the ``doc/source/examples/`` directory is a Jupyter notebook.
+They are rendered in the user guide on the "Examples" page, and
```
Comment (RMeli) [inline doc/source/contributing_docs.rst]: Worth linking to the page?

File: doc/source/contributing_docs.rst [hunk 6]
```diff
@@ -83,6 +75,133 @@ Using ``conda`` or similar (``miniconda``, ``mamba``, ``micromamba``), create a
...
+are also listed in the "Analysis" section of the user guide.
```
Comment (RMeli) [inline doc/source/contributing_docs.rst]: Worth linking to the section?

**Review by RMeli — COMMENTED** [review]
Partial review.

File: doc/source/contributing_docs.rst [hunk 7]
```diff
@@ -169,52 +284,47 @@ environmental variable like this::
 Adding changes to the UserGuide
 ===============================
 
-As with the code, :ref:`commit and push <adding-code-to-mda>` your code to GitHub. Then :ref:`create a pull request <create-a-pull-request>`. The only test run for the User Guide is: that your file compile into HTML documents without errors. As usual, a developer will review your PR and merge the code into the User Guide when it looks good.
+As with the code, :ref:`commit and push <adding-code-to-mda>` your code to GitHub.
+Then :ref:`create a pull request <create-a-pull-request>`.
+The only test run for the User Guide is: that your file compile into HTML documents without errors.
```
Comment (RMeli) [inline doc/source/contributing_docs.rst]:
```suggestion
The only test run for the User Guide is that your file compile into HTML documents without errors.
```
?

File: doc/source/contributing_docs.rst [hunk 8]
```diff
@@ -169,52 +284,47 @@ environmental variable like this::
...
+As usual, a developer will review your PR and merge the code into the User Guide when it looks good.
```
Comment (RMeli) [inline doc/source/contributing_docs.rst]:
```suggestion
As usual, your PR will be reviewed and merged into the User Guide when it looks good.
```
Just to be a bit more open about who can/will do this? I.e. a non-developer but technical writer, etc.

File: doc/source/contributing_docs.rst [hunk 9]
```diff
@@ -169,52 +284,47 @@ environmental variable like this::
...
 
-It is often difficult to review Jupyter notebooks on GitHub, especially if you embed widgets and images. One way to make it easier on the developers who review your changes is to build the changes on your forked repository and link the relevant sections in your pull request. To do this, create a ``gh-pages`` branch and merge your new branch into it.
+If you have issues building your documentation locally, pull requests
+create preview documentation on ReadTheDocs, which you can use to check renders.
```
Comment (RMeli) [inline doc/source/contributing_docs.rst]:
```suggestion
If you have issues building your documentation locally, opening a pull request
creates preview documentation on ReadTheDocs, which you can use to check renders.
```
I think it reads a bit more clearly (at least to me)?

File: doc/source/contributing_docs.rst (hunk 2 above)
Comment (RMeli) [inline doc/source/contributing_docs.rst]: Worth adding a link to pre-commit?

**Comment by RMeli** [comment]
@lilyminium is this PR still relevant?

File: doc/source/contributing_docs.rst [hunk 10]
```diff
@@ -83,6 +75,133 @@ Using ``conda`` or similar (``miniconda``, ``mamba``, ``micromamba``), create a
...
+""""""""""""""""""""""
+
+These are the files that end in ``.rst`` and contain plain text.
+reStructuredText is a markup language that is used to write
+documentation for Sphinx in Python.
+The `reStructuredText Primer <https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html>`_
+is a good place to start if you are unfamiliar with the syntax.
+
+The user guide also makes use of a number of Sphinx extensions to ensure that the code examples are always up-to-date.
+One example is the `ipython directive <http://matplotlib.org/sampledoc/ipython_directive.html>`__.
```
Comment (orbeckst) [inline doc/source/contributing_docs.rst]:
```suggestion
One example is the `ipython directive <https://matplotlib.org/sampledoc/ipython_directive.html>`__.
```

**Review by orbeckst — APPROVED** [review]
All looks good to me. 

Even if it's not perfect — there are some suggestions that could be addressed — it would be much more useful merged than lingering as a stale PR. I suggest to merge it asap.

File: doc/source/contributing_docs.rst [hunk 11]
```diff
@@ -169,52 +284,49 @@ environmental variable like this::
...
 
+The below sections are optional, but may be helpful for more advanced users.
 
-Now you can build your pages with the ``make github`` macro in the ``UserGuide/doc/`` directory, which builds the files and copies them to the top level of your directory.
+Using pre-commit hooks
+""""""""""""""""""""""
 
-.. code-block:: bash
+Manually editing files can often lead to small inconsistencies: a whitespace here, a missing carriage return there.
+A tool called pre-commit can be used to automatically fix these problems, before a git commit is made.
```
Comment (orbeckst) [inline doc/source/contributing_docs.rst]:
```suggestion
A tool called `pre-commit <https://pre-commit.com/>`_ can be used to automatically fix these problems, before a git commit is made.
```

**Comment by orbeckst** [comment]
@lilyminium if you can confirm that the PR is still relevant then let's merge it.

**Comment by orbeckst** [comment]
Once https://github.com/osmart/hole2/pull/19 is merged, it will be sufficient to just link to the hole2 GH README for installation instructions and we can keep our UG lighter [dd8000e](https://github.com/MDAnalysis/UserGuide/pull/330/commits/dd8000e8ff46bb993beb4c16e743cd6a1586b1c3)

File: doc/source/contributing_docs.rst (hunk 6 above)
Comment (lilyminium) [reply doc/source/contributing_docs.rst]: Added links!

File: doc/source/contributing_docs.rst (hunk 3 above)
Comment (lilyminium) [reply doc/source/contributing_docs.rst:351]: Added a link

**Comment by lilyminium** [comment]
Yep, this is still relevant, and I just made some minor changes in response to reviews -- thanks for having a look at this everyone!

File: doc/source/contributing_docs.rst [hunk 12]
```diff
@@ -83,6 +66,136 @@ Using ``conda`` or similar (``miniconda``, ``mamba``, ``micromamba``), create a
...
+  for automatically generated documentation
+* Text files (``.txt``) which contain the automatically generated documentation.
+  These should not be edited directly as changes will not be included.
+  These are largely tables and lists of topology attributes for readers, writers,
+  and parsers.
+
+
+What file to edit
+"""""""""""""""""
+The structure of the user guide is a bit convoluted!
```
Comment (RMeli) [inline doc/source/contributing_docs.rst]:
```suggestion
```

File: doc/source/contributing_docs.rst [hunk 13]
```diff
@@ -169,52 +273,65 @@ environmental variable like this::
...
 
-.. code-block:: bash
 
-    git checkout gh-pages
-    git merge origin/my_branch
-    cd doc/
-    make github
+Then :ref:`create a pull request <create-a-pull-request>`.
+The only test run for the User Guide is: that your file compile into HTML documents without errors.
+As usual, a developer will review your PR and merge the code into the User Guide when it looks good.
```
Comment (RMeli) [inline doc/source/contributing_docs.rst]: Repetition of the section above. Is this intentional?

**Review by RMeli — APPROVED** [review]

File: doc/source/contributing_docs.rst (hunk 12 above)
Comment (orbeckst) [reply doc/source/contributing_docs.rst]: I agree – never apologize for supplying documentation!

File: doc/source/contributing_docs.rst (hunk 13 above)
Comment (lilyminium) [reply doc/source/contributing_docs.rst]: No, oops -- removed as an accidental copy-paste. Thanks for the catch!

**Comment by orbeckst** [comment]
I am merging as it's (1) approved and (2) a big improvement over what we have. If there are any updates to be made, we can do them incrementally.

**Pull request merged**